### PR TITLE
Dynamic Excel Imports

### PR DESCRIPTION
**Pull request by inviscid**
This removed some of the list and dict comprehensions because it helped with understanding the logic flow a bit better

### CODE REVIEW

File: plaidcloud/utilities/frame_manager.py [hunk 1]
```diff
@@ -2434,43 +2460,73 @@ def excel_to_csv_xlrd(excel_file_name, csv_file_name, sheet_name='sheet1', clean
...
+                                row.append(int(c.value)) # This appears to be an int
+                            else:
+                                row.append(get_formatted_number(c.value)) # some other type of number
+                        elif dtype == 'boolean':
+                            row.append(c.value)
+                        elif dtype == 'datetime':
+                            if xlrd.xldate_as_tuple(c.value, datemode)[0] == 0:
+                                row.append(null_value)
+                            else:
+                                row.append(datetime.time(*xlrd.xldate_as_tuple(c.value, datemode)[:3]).isoformat())
```
Comment (rad-pat) [inline plaidcloud/utilities/frame_manager.py:2518]: This looks setup the wrong way around. Should prob read something like:
```
                            date_tuple = xlrd.xldate_as_tuple(c.value, datemode)
                            if date_tuple[0] == 0:
                                row.append(datetime.time(*date_tuple[:3]).isoformat())
                            else:
                                row.append(datetime.datetime(*date_tuple).isoformat())
```

File: plaidcloud/utilities/frame_manager.py [hunk 2]
```diff
@@ -2397,34 +2401,56 @@ def excel_to_csv_xlrd(excel_file_name, csv_file_name, sheet_name='sheet1', clean
...
-                            column_name = 'FALSE'
+                            column_name = source_name
+                        
+                        # logger.info(f'----- Raw Column name prior to compliance check {column_name}')
+
+                        if column_name[0:1] in string.digits:
+                            column_name = f'_{column_name}'
+                        column_name = ''.join([c for c in column_name if c not in invalid_characters]) # Remove invalid characters
+                        column_name = column_name[:63] # Truncate to max length
+                        data_dtype = dtype_from_excel(sh.cell(rownum + 1, col).ctype) # go to next row to determine dytype of the data
```
Comment (rad-pat) [inline plaidcloud/utilities/frame_manager.py:2444]: What if there is no next row?

File: plaidcloud/utilities/frame_manager.py (hunk 1 above)
Comment (rad-pat) [reply plaidcloud/utilities/frame_manager.py:2518]: Actually, looking back up in the code, there is a `xlrd.xldate_as_datetime` method

File: plaidcloud/utilities/frame_manager.py [hunk 3]
```diff
@@ -2397,34 +2401,56 @@ def excel_to_csv_xlrd(excel_file_name, csv_file_name, sheet_name='sheet1', clean
...
                     # See if this column name is a date type
-                    column_name_dtype = dtype_from_excel(sh.cell(skip_rows, col).ctype)
-                    if column_name_dtype == 'timestamp':
-                        # Need to convert this from an Excel date stamp to human readable form
-                        column_date_name = xlrd.xldate_as_datetime(sh.cell(skip_rows, col).value, 0).date().isoformat()
-                        column_name = f'_{column_date_name}' # column names can't start with numbers so we prefix with _
-                    elif column_name_dtype == 'boolean':
-                        if sh.cell(skip_rows, col).value:
-                            column_name = 'TRUE'
+                    column_name_dtype = dtype_from_excel(sh.cell(rownum, col).ctype)
```
Comment (rad-pat) [inline plaidcloud/utilities/frame_manager.py:2419]: Maybe grab the cell here, like below where it reads data `c = sh.cell(rownum, col_pos)`, then the code reads similar?`c.ctype, c.value`

File: plaidcloud/utilities/frame_manager.py [hunk 4]
```diff
@@ -2434,43 +2460,73 @@ def excel_to_csv_xlrd(excel_file_name, csv_file_name, sheet_name='sheet1', clean
...
+                    column_information.append({
+                        'source': source_name,
+                        'target': column_name,
+                        'dtype': data_dtype 
+                    })
+
+                # logger.info(f'Column information for Excel Import: {column_information}')
                 wr.writerow(header_columns)
+                column_count = len(column_information)
+                continue
```
[the remark this answers is not in the document]
Comment (inviscid) [reply plaidcloud/utilities/frame_manager.py:2483]: No.  Just being overly careful.  Removed

**Pull request merged**